dascore/core/spool.py

```diff
@@ -309,6 +316,37 @@ def select(self, **kwargs) -> Self:
         )
         return out
 
+    @compose_docstring(doc=BaseSpool.sort.__doc__)
+    def sort(self, attribute) -> Self:
+        """Sort the Spool based on a specific attribute."""
```

Comment:
The `compose_docstring` decorator allows you to reuse strings in the documentation. You don't appear to be using it here though. So perhaps you want this?

```python
@compose_docstring(doc=BaseSpool.sort.__doc__)
def sort(self, attribute) -> Self:
    """
    {doc}
    """
```

Also, We should add more to the docs explaining the parameters and such. See note above.